internal/controller/pkg/revision/hook.go

```diff
@@ -112,12 +111,45 @@ func (h *ProviderHooks) Pre(_ context.Context, pkg runtime.Object, pr v1.Package
 
 	// TODO(hasheddan): update any status fields relevant to package revisions.
 
+	if pr.GetDesiredState() == v1.PackageRevisionActive {
+		return h.ensurePrerequisites(ctx, pkgProvider, pr)
```

Comment:
> we need to setup the tls server secret before the APIEstablisher try to use it to inject the certificate into CRDs/Webhooks.

How was it working before? Or was it working at all? 🤔

Reply:
A code comment with the above information would be helpful for posterity.

Reply:
It was working before because we were using the only webhook we had, the same crossplane was using, which was using a pretty wide *.<namespace>. I'll add a comment 👍